### PR TITLE
Use Makefile to build instead of bash scripts

### PR DESCRIPTION
**Pull request by sdemos**
Makes building more predictable and cleans up the root directory.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	BUILD_VERBOSE = $(V)
+endif
+
+ifeq ($(BUILD_VERBOSE),1)
+	Q =
+else
+	Q = @
+endif
+
+export GO15VENDOREXPERIMENT:=1
```
Comment (dgonyeo) [inline Makefile]: tiny nit: I think we can drop support for go 1.5

Comment (sdemos) [inline Makefile]: I agree, but for now the `.travis.yml` file includes go 1.5.4 in the language version matrix. Will probably drop support for that soon.

Comment (dghubble) [inline Makefile]: Was about to comment on that. Let's drop 1.5 and add 1.8 in a followup and remove this env var then

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+VERSION=$(shell git describe --dirty)
+REPO=github.com/coreos/locksmith
+LD_FLAGS="-w -s -extldflags -static"
```
Comment (dghubble) [inline Makefile]: We don't have other projects setting these, but looks like they're just from before

**Comment by dghubble** [comment]
There's a fairly common test script we might switch to in a followup instead of every project writing their own special sauce.

**Comment by sdemos** [comment]
I was going to delete that test script and just do it the way cluo does it (test target with `go test -v $(REPO)/...`) but there is a `cover` script that relies on it being tested this way. Maybe @crawford can give more insight into what that is and if it is still needed? If we can drop the `cover` script then I have no problems with using the more common approach.

**Comment by dghubble** [comment]
I think cluo isn't the greatest example here. Its not going linting, vet, fmt, cover, check, or anything really.

**Comment by sdemos** [comment]
In that case, I'll merge this and we can update the testing later.

**Pull request merged**